packages/react-core/src/async-loader.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+export interface Container extends Window {
+  init: (module: any) => void;
+}
+
+declare function __webpack_init_sharing__(scope: string): void;
+declare let __webpack_share_scopes__: any;
+
+const DefaultErrorComponent: React.ComponentType<any> = () => {
+  return <span>Error while loading component!</span>;
+};
+
+export function loadComponent(scope: string, module: string, ErrorComponent: React.Component) {
```

Comment:
There should be `ErrorComponent?:` to make it optional or you can initialize it directly:

```ts
export function loadComponent(scope: string, module: string, ErrorComponent: React.Component = DefaultErrorComponent)
```

Reply:
I was unable to cast `React.Component` to `DefaultComponent` and assign in function call for some reason.

Reply:
I think it should React.ComponentType not Component